diff --git a/multiselect/index.mjs b/multiselect/index.mjs
index 8ba9a8d407b3c6324561eb18510a9a73f9543262..2d3f2b0d2de3f3e9e9f18ee48eab6e10cc5022da 100644
--- a/multiselect/index.mjs
+++ b/multiselect/index.mjs
@@ -746,7 +746,8 @@ var script = {
       }
     },
     isOutsideClicked: function isOutsideClicked(event) {
-      return !(this.$el.isSameNode(event.target) || this.$el.contains(event.target) || this.overlay && this.overlay.contains(event.target));
+      const path = event.composedPath();
+      return !(path.includes(this.$el) || (this.overlay && path.includes(this.overlay)));
     },
     getLabelByValue: function getLabelByValue(value) {
       var _this9 = this;
