diff --git a/multiselect/index.mjs b/multiselect/index.mjs
index bbe16fb37e386775a93158133e04db6619d11ca7..592ee6c5347269ef6a058f74740d2a580ec75208 100644
--- a/multiselect/index.mjs
+++ b/multiselect/index.mjs
@@ -747,7 +747,8 @@ var script = {
       }
     },
     isOutsideClicked: function isOutsideClicked(event) {
-      return !(this.$el.isSameNode(event.target) || this.$el.contains(event.target) || this.overlay && this.overlay.contains(event.target));
+      const path = event.composedPath();
+      return !(path.includes(this.$el) || (this.overlay && path.includes(this.overlay)));
     },
     getLabelByValue: function getLabelByValue(value) {
       var _this9 = this;
